cabal-install/Distribution/Client/ProjectConfig.hs

```diff
@@ -879,7 +888,44 @@ readSourcePackage verbosity (ProjectPackageLocalDirectory dir cabalFile) = do
       packageSource        = LocalUnpackedPackage (root </> dir),
       packageDescrOverride = Nothing
     }
-readSourcePackage _verbosity _ =
+
+readSourcePackage verbosity distDirLayout (ProjectPackageRemoteRepo repo) = do
+    let
+      scmSrcDir  = "scm-" ++ showHashValue (hashSourceRepo repo)
+      destDir = distUnpackedSrcRootDirectory distDirLayout </> scmSrcDir
+
+      subdir  = fromMaybe "." (repoSubdir repo)
+      pkgdir  = destDir </> subdir
+
+      -- we only need this for error messages
+      repoloc = fromMaybe "" (repoLocation repo)
+
+    repoExists <- liftIO $ doesDirectoryExist destDir
```

Comment:
Ok, here is where we will need to think about the incremental case. Notice that we are in a `Rebuild` monad. Currently, this `readSourcePackage` will get re-run every time the cabal.project file changes. That could be adjusted so we only re-run when the input information for the source repo changes.

BTW, notice that for the local file case we do a `monitorFiles [monitorFileHashed cabalFile]` to force this code to get re-run when the user edits that file. For source repos we don't need to worry about the user editing things under us I think.